manager/manager.go

```diff
@@ -1004,27 +1009,29 @@ func (m *Manager) becomeLeader(ctx context.Context) {
 	// If DefaultAddrPool is null, Read from store and check if
 	// DefaultAddrPool info is stored in cluster object
 	// If VXLANUDPPort is 0, read it from the store - cluster object
-	if m.config.NetworkConfig == nil || m.config.NetworkConfig.DefaultAddrPool == nil || m.config.NetworkConfig.VXLANUDPPort == 0 {
+	if m.config.DefaultAddrPool == nil || m.config.VXLANUDPPort == 0 {
 		var cluster *api.Cluster
 		s.View(func(tx store.ReadTx) {
 			cluster = store.GetCluster(tx, clusterID)
 		})
 		if cluster.DefaultAddressPool != nil {
-			if m.config.NetworkConfig == nil {
-				m.config.NetworkConfig = &cnmallocator.NetworkConfig{}
-			}
-			m.config.NetworkConfig.DefaultAddrPool = append(m.config.NetworkConfig.DefaultAddrPool, cluster.DefaultAddressPool...)
-			m.config.NetworkConfig.SubnetSize = cluster.SubnetSize
+			m.config.DefaultAddrPool = append(m.config.DefaultAddrPool, cluster.DefaultAddressPool...)
+			m.config.SubnetSize = cluster.SubnetSize
 		}
 		if cluster.VXLANUDPPort != 0 {
-			if m.config.NetworkConfig == nil {
-				m.config.NetworkConfig = &cnmallocator.NetworkConfig{}
-			}
-			m.config.NetworkConfig.VXLANUDPPort = cluster.VXLANUDPPort
+			m.config.VXLANUDPPort = cluster.VXLANUDPPort
 		}
 	}
 
-	m.allocator, err = allocator.New(s, m.config.PluginGetter, m.config.NetworkConfig)
+	// Toss all this info into a netCfg, so that the allocator can do its
+	// thing.
+	netCfg := &cnmallocator.NetworkConfig{
```

Comment:
why are you creating a new config instead of using `m.config.NetworkConfig`?

Reply:
^^ @dperny good point; especially in light of;

> The long-pending long-awaited allocator rewrite on the new-allocator branch removes cnmallocator, and doing this refactoring in an earlier commit will simplify that operation greatly.